### PR TITLE
Stabilize feature `saturating_div` for rust 1.58.0

### LINKED ISSUE #89381: Tracking Issue for `saturating_div`
The feature gate for the issue is `#![feature(saturating_div)]`.

### About tracking issues

This tracking issue tracks the progress of the `saturating_div` feature, which shall provide a saturating division for integer types. It is part of the ongoing effort to stabilize the `Saturating` type #88624

### Unresolved Questions

None. The implementation has been discussed [here](https://github.com/rust-lang/rust/pull/87921#issuecomment-899357720) and [here](https://github.com/rust-lang/rust/pull/87921#discussion_r691888556).

### Steps

- [x] implement (#87921)
- [ ] Stabilize `saturating_div` (#88624)

[stabilization-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#stabilization-pr
[doc-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#documentation-prs

### Implementation history

 - Implementation #87921
   - Adds `fn saturating_div` and `feature(saturating_div)`

### PR DESCRIPTION
**Pull request by kellerkindt**
The tracking issue is #89381

This seems like a reasonable simple change(?). The feature `saturating_div` was added as part of the ongoing effort to implement a `Saturating` integer type (see #87921). The implementation has been discussed [here](https://github.com/rust-lang/rust/pull/87921#issuecomment-899357720) and [here](https://github.com/rust-lang/rust/pull/87921#discussion_r691888556). It extends the list of saturating operations on integer types (like `saturating_add`, `saturating_sub`, `saturating_mul`, ...) by the function `fn saturating_div(self, rhs: Self) -> Self`.

The stabilization of the feature `saturating_int_impl` (for the `Saturating` type) needs to have this stabilized first.

Closes #89381

### CODE REVIEW

File: library/core/src/num/uint_macros.rs [hunk 1]
```diff
@@ -1049,20 +1049,16 @@ macro_rules! uint_impl {
...
         /// ```should_panic
-        /// #![feature(saturating_div)]
-        ///
         #[doc = concat!("let _ = 1", stringify!($SelfT), ".saturating_div(0);")]
         ///
         /// ```
-        #[unstable(feature = "saturating_div", issue = "87920")]
-        #[rustc_const_unstable(feature = "saturating_div", issue = "87920")]
+        #[stable(feature = "saturating_div", since = "1.56.0")]
+        #[rustc_const_stable(feature = "saturating_div", since = "1.56.0")]
```
Comment (camelid) [inline library/core/src/num/uint_macros.rs:1061]: 1.56.0 is about to be promoted to beta (within the next few days), so this will probably land in 1.57.0.

**Comment by camelid** [comment]
r? rust-lang/libs

**Comment by joshtriplett** [comment]
:+1: for stabilizing this, but I agree that it doesn't seem likely to hit 1.56. r=me as long as the `since` version matches the version it's actually going to land in though.

**Comment by kellerkindt** [comment]
I see. I pushed it back to stabilize in 1.57 :)

**Comment by joshtriplett** [comment]
FCP for stabilizing `saturating_div`: https://github.com/rust-lang/rust/issues/87920#issuecomment-912943836

**Comment by jhpratt** [comment]
I think the `#[rustc_const_stable]` attribute can be omitted when it's being stabilized at the same time.

**Review by JohnTitor — APPROVED** [review]

**Review by Vic23M — APPROVED** [review]

**Review by Vic23M — APPROVED** [review]

**Pull request merged**